src/components/ui/CustomModal.tsx

```diff
@@ -49,13 +52,18 @@ const CustomModal: FC<Props> = ({
     handleClose();
   };
   const handleOk = (event: MouseEvent<HTMLButtonElement>) => {
-    if (onOk) onOk(event);
-    handleClose();
+    if (onOk) {
+      onOk(event);
+      handleClose();
+    }
   };
   return (
     <Fragment>
       {buttonComponent ? (
-         React.cloneElement(buttonComponent, { label: buttonText, onClick: handleOpen })
+        React.cloneElement(buttonComponent, {
```

Comment:
Is cloning necessary?

Reply:
Yes, I am passing a buttonComponent like this <CustomButton id="btn" />, and for opening the modal on this button click we need to pass the onClick method that's why I have cloned it.